File: algorithms/172.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int trailingZeroes(int n)
    {
        int count = 0;
        for ( long long i = 5; n/i; i *= 5 )
            count += n/i;
        return count;
    }
};

int main()
{
    Solution solution;
    //cout << solution.trailingZeroes(1808548329) << endl;
    cout << solution.trailingZeroes(15) << endl;
    return 0;
}

File: algorithms/python/172.py
class Solution(object):
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        cnt5 = 0
        base = 5
        while base <= n:
            cnt5 += n // base
            base *= 5
        return cnt5
